python/ambassador/ir/irhealthchecks.py

```diff
@@ -0,0 +1,256 @@
+                    self.post_error(
+                        f"IRHealthChecks: health_check.http: field must be an object, found {http_health_check}. Ignoring health-check {hc}",
+                        log_level=logging.ERROR,
+                    )
+                    continue
+
+                # Make sure we have a path
+                path = http_health_check.get("path", None)
+                if path is None:
+                    self.post_error(
+                        f"IRHealthChecks: health_check.http.path is a required field. Ignoring health-check: {hc}",
+                        log_level=logging.ERROR,
+                    )
+                    continue
+                http_mapper: Dict[str, Any] = {"path": path}
+
+                # Process header add/remove operations
+                request_headers_to_add = http_health_check.get("add_request_headers", None)
+                if request_headers_to_add is not None:
+                    if isinstance(request_headers_to_add, list):
+                        self.post_error(
+                            f"IRHealthChecks: add_request_headers must be a dict of header:value pairs. Ignoring field for health-check: {hc}",
+                            log_level=logging.ERROR,
+                        )
+                    addHeaders = self.generate_headers_to_add(request_headers_to_add)
+                    if len(addHeaders) > 0:
+                        http_mapper["request_headers_to_add"] = addHeaders
+                request_headers_to_remove = http_health_check.get("remove_request_headers", None)
+                if request_headers_to_remove is not None:
+                    if not isinstance(request_headers_to_remove, list):
+                        self.post_error(
+                            f"IRHealthChecks: remove_request_headers must be a list. Ignoring field for health-check: {hc}",
+                            log_level=logging.ERROR,
+                        )
+                    else:
+                        http_mapper["request_headers_to_remove"] = request_headers_to_remove
+
+                host = http_health_check.get("hostname", None)
+                if host is not None:
+                    http_mapper["host"] = host
+
+                # Process the expected statuses
+                expected_statuses = http_health_check.get("expected_statuses", None)
+                if expected_statuses is not None:
+                    validStatuses = []
+                    for statusRange in expected_statuses:
+                        minCode = int(statusRange["min"])
+                        maxCode = int(statusRange["max"])
+                        # We add one to the end code because by default Envoy expects the start of the range to be
+                        # inclusive, but the end of the range to be exclusive. Lets just make both inclusive for simplicity.
+                        maxCode += 1
+                        if minCode < 100 or minCode > 600:
```

Comment:
I'm confused why these checks need to be here, instead of trusting the CRD validations.

Reply:
removed all the ones that were not necessary for the linting type checks